### PR TITLE
refactor preflashed credentials into separate build target to eliminate possibility of commiting credentials in repo

### PR DESCRIPTION
**Pull request by positavi**

### CODE REVIEW

File: src/credentials_template.h [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
+ *  only applicable for [env:demo-unit-preloaded]
+ *
+ *  to use this file, copy this template to "credentials.h" and fill in the applicable values.  
+ *  the list below is non-exhaustive and additional options can be added as needed.
+ *  this will override anything in the file system on every startup.
+*/
+
+
 void set_preferences(Preferences preferences);
 bool preload_credentials = false;
```
Comment (brandon-wada) [inline src/credentials_template.h]: We can drop this line too now

File: src/credentials_template.h [hunk 2]
```diff
@@ -1,5 +1,15 @@
 #include <Preferences.h>
 
+
+/*  template file for preloading credentials instead of reading from the file system 
+ *  only applicable for [env:demo-unit-preloaded]
+ *
+ *  to use this file, copy this template to "credentials.h" and fill in the applicable values.  
+ *  the list below is non-exhaustive and additional options can be added as needed.
+ *  this will override anything in the file system on every startup.
+*/
```
Comment (brandon-wada) [inline src/credentials_template.h:10]: Should we add the template file to the .gitignore too? Add one more layer of protection requiring someone to supercede the ignore before accidentally checking in credentials in this file

**Review by brandon-wada — APPROVED** [review]
Looks good, I have 2 one-line suggestions

File: src/credentials_template.h (hunk 2 above)
Comment (positavi) [reply src/credentials_template.h:10]: AFAIK if the file is in the repo, the .gitignore won't do quite what we want.

File: src/credentials_template.h (hunk 1 above)
Comment (positavi) [reply src/credentials_template.h]: good catch, thanks!

**Pull request merged**